File: p020.py
import time

def compute():
    val = 1

    for i in range(2, 101):
        val *= i
        while val % 10 == 0:
            val //= 10

    return sum([int(d) for d in str(val)])

if __name__ == '__main__':
    print(compute())

File: p020.cpp
#include <iostream>
#include <math.h>

using namespace std;

#define SIZE 180

int compute() {
    int num[SIZE] = {};
    num[0] = 1;

    for (int i = 2; i < 100; i++) {
        int carry = 0;
        for (int j = 0; j < SIZE; j++) {
            int digit_with_carry = num[j] * i + carry;

            num[j] = digit_with_carry % 10;
            carry = digit_with_carry / 10;
        }
    }

    int sum = 0;
    for (int d: num) sum += d;
    return sum;
}

int main() {
    cout << compute() << endl;
    return 0;
}
